example/lib/main.dart

```diff
@@ -78,108 +99,175 @@ class _MyAppState extends State<MyApp> implements RoomHandler {
-                            return 'A user ID is required';
-                          }
-                          return null;
-                        }),
-                  ),
-                  Padding(
-                    padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 16),
-                    child: TextFormField(
-                      controller: _tokenController,
-                      decoration: const InputDecoration(
-                        border: OutlineInputBorder(),
-                        // Announce validation errors (https://github.com/flutter/flutter/issues/99715).
-                        helperText: kIsWeb ? ' ' : null,
-                        labelText: 'Token',
-                      ),
-                      enabled: !_isInCall,
-                      validator: (String? value) {
-                        if (value == null || value.isEmpty) {
-                          return 'A token is required';
-                        }
-                        return null;
-                      },
-                    ),
-                  ),
-                  ElevatedButton(
-                    onPressed: _isInCall ? _hangUp : _callAira,
-                    style: ElevatedButton.styleFrom(
-                      primary: _isInCall ? Theme.of(context).errorColor : Theme.of(context).primaryColor,
-                    ),
-                    child: Text(_isInCall ? 'End Call' : 'Call an Aira Agent'),
-                  ),
-                  Visibility(
-                    visible: false,
-                    child: RTCVideoView(_remoteRenderer),
-                  ),
-                ],
+      body: Center(
+        child: Padding(
+          padding: const EdgeInsets.all(16),
+          child: LayoutBuilder(builder: (BuildContext context, BoxConstraints viewportConstraints) {
+            return SingleChildScrollView(
+              child: ConstrainedBox(
+                constraints: BoxConstraints(
+                  maxWidth: 640 - 32,
+                  minHeight: viewportConstraints.maxHeight - 32,
+                ),
+                child: IntrinsicHeight(
+                  child: _isInCall ? _inCallWidget : _callSetupWidget,
+                ),
               ),
+            );
+          }),
+        ),
+      ),
+    );
+  }
+
+  Widget get _inCallWidget {
+    return Column(
+      children: <Widget>[
+        Visibility(
+          visible: false,
+          child: RTCVideoView(_remoteRenderer),
+        ),
+        Visibility(
+          visible: _isMessagingEnabled,
+          child: _messagingWidget,
+        ),
+        const SizedBox(height: 16),
+        ElevatedButton(
+          onPressed: _hangUp,
+          style: ElevatedButton.styleFrom(primary: Theme.of(context).errorColor),
+          child: const Text('End Call'),
+        ),
+      ],
+    );
+  }
+
+  Widget get _messagingWidget {
+    return Row(
+      children: <Widget>[
+        Expanded(
```

Comment:
Without `Expanded`, sticking `TextFormField` in a `Row` causes layout issues.

https://stackoverflow.com/a/56923281